libtiledbvcf/src/dataset/tiledbvcfdataset.cc

```diff
@@ -716,6 +718,21 @@ void TileDBVCFDataset::load_field_type_maps_v4(const bcf_hdr_t* hdr) const {
     hdr = hdrs.begin()->second.get();
   }
 
+  if (add_iaf) {
+    // TODO: do something better than promoting this pointer type; perhaps the
+    // header should be duplicated and later modified
```

Comment:
Do we need to address this `TODO`? If not, remove the comment.

Reply:
I wanted to leave that TODO, because altering a pointer to a const is still a problem there.